src/plugins/unified_search/public/query_string_input/query_bar_top_row.tsx

```diff
@@ -527,6 +528,7 @@ export const QueryBarTopRow = React.memo(
     }
 
     function renderDataViewsPicker() {
+      if (props.customDataViewPicker) return props.customDataViewPicker;
```

Comment:
I would prefer this check to not happen here but happen here instead https://github.com/elastic/kibana/blob/main/src/plugins/unified_search/public/query_string_input/query_bar_top_row.tsx#L677


In my mind this is a custom element which is rendered in the position of the dataview picker. In your example I see a saved searches picker (not dataviews) and can be anything else. So I think:
1. Should not be on the renderDataViewPicker function but be a different function instead
2. Change the name, customDataViewPicker indicates to me that I have a dataview picker which just uses a different layout. But this is something else.

Reply:
Makes sense to me -- I renamed the prop to `dataViewPickerOverride` instead and moved the check to where you suggested: https://github.com/elastic/kibana/pull/158603/commits/fc8d15e0c6a0c41b0bf62d1d99740be8280a625f. Let me know if that seems good!